include/ideep/operators/inner_product.hpp

```diff
@@ -259,14 +270,19 @@ struct inner_product_backward_data : public dnnl::inner_product_backward_data {
     auto pd = primitive_desc(
         {diff_src_desc, weights_desc, diff_dst_desc}, aengine, forward_hints);
 
+    // reorder diff_dst(grad_y), weights, diff_src(grad_x) if needed
     auto expected_diff_dst = diff_dst.reorder_if_differ_in(pd.diff_dst_desc());
     auto expected_weights = weights_.reorder_if_differ_in(pd.weights_desc());
-    diff_src.reinit_if_possible(pd.diff_src_desc());
+    // see [Notes output buffer]
+    if (diff_src.is_empty())
+      diff_src.reinit_if_possible(pd.diff_src_desc());
```

Comment:
same to use diff_src.init(pd.diff_src_desc());

Reply:
done